src/main/java/sirius/biz/jobs/batch/file/VirtualFileExtractionJob.java

```diff
@@ -0,0 +1,214 @@
+    private static final String TARGET_PATH = "targetPath";
+
+    @Override
+    protected void execute(ProcessContext process) throws Exception {
+        VirtualFile sourceFile = process.require(SOURCE_PARAMETER);
+        Optional<VirtualFile> destinationDirectory = DESTINATION_PARAMETER.get(process.getContext());
+        boolean shouldOverwriteExisting = process.require(OVERWRITE_EXISTING_FILES_PARAMETER);
+
+        // by default we'll use the files directory to extract to
+        final VirtualFile targetDirectory =
+                destinationDirectory.orElseGet(() -> vfs.resolve(sourceFile.parent().path()));
+
+        sourceFile.tryDownload()
+                  .ifPresent(getArchiveExtractCallback(process, shouldOverwriteExisting, targetDirectory));
+    }
+
+    @Nonnull
+    private Consumer<FileHandle> getArchiveExtractCallback(ProcessContext process,
+                                                           boolean shouldOverwriteExisting,
+                                                           VirtualFile targetDirectory) {
+        return fileHandle -> {
+            File tempFile = fileHandle.getFile();
+            if (!TaskContext.get().isActive()) {
+                Files.delete(tempFile);
+                return;
+            }
+
+            try {
+                ArchiveHelper.extract(tempFile,
+                                      null,
+                                      (status, data, filePath, filesProcessedSoFar, bytesProcessedSoFar, totalBytes) -> {
+                                          if (!TaskContext.get().isActive()) {
+                                              return false;
+                                          }
+                                          VirtualFile targetFile =
+                                                  vfs.resolve(vfs.makePath(targetDirectory.path(), filePath));
+                                          if (targetFile.exists() && !shouldOverwriteExisting) {
+                                              process.log(ProcessLog.info()
+                                                                    .withMessage(NLS.fmtr(
+                                                                            "VirtualFileExtractionJob.skippingOverwrite")
+                                                                                    .set(TARGET_PATH, targetFile.path())
+                                                                                    .format()));
+                                              return false;
+                                          }
+
+                                          uploadFile(process, shouldOverwriteExisting, data, filePath, targetFile);
+                                          updateState(status, filesProcessedSoFar, bytesProcessedSoFar, totalBytes);
+
+                                          return true;
+                                      });
```

Comment:
Am Besten ne Funktion raus ziehen

Reply:
Stimme mich hierzu mit @andyHa ab, welche Möglichkeiten hier Sinn machen. Bspw. wäre ein Consumer denkbar, der optional beim Entpacken übergeben werden kann. Das Callback hat aktuell hauptsächlich Progress-Funktionalität um dem Aufrufer den Status mitteilen zu können. Rausziehen der Stoplogik dürfte hingegen eher schwierig werden, da diese aktuell ja auch vom Callback mitgehandled wird.

Reply:
kann man die kilometer lambda net einfach in eine funktion rausziehen und bissle zerlegen?

Reply:
Wie besprochen zerlegt und sauber als Predicate durchgereicht.